drivers/ds75lx/ds75lx.c

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2019 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ds75lx
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the DS75LX temperature sensor.
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "xtimer.h"
+
+#include "ds75lx.h"
+#include "ds75lx_internals.h"
+#include "ds75lx_params.h"
+#include "periph/i2c.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+#define DEV_I2C      (dev->params.i2c)
+#define DEV_ADDR     (dev->params.addr)
+
+static const uint8_t _max_conversion_times[4] = {
+    DS75LX_MAX_CONVERSION_25,
+    DS75LX_MAX_CONVERSION_50,
+    DS75LX_MAX_CONVERSION_100,
+    DS75LX_MAX_CONVERSION_250,
+};
+
+static int _set_configuration_bit(const ds75lx_t *dev, uint8_t bit, bool enable)
+{
+    /* Acquire exclusive access */
+    i2c_acquire(DEV_I2C);
+
+    uint8_t config;
+    if (i2c_read_reg(DEV_I2C, DEV_ADDR, DS75LX_REG_CONFIGURATION, &config, 0) < 0) {
+        DEBUG("[ds75lx] error reading configuration register\n");
+        /* Release I2C device */
+        i2c_release(DEV_I2C);
+
+        return -DS75LX_ERR_I2C;
+    }
+
+    DEBUG("[ds75lx] initial configuration register value: 0x%02X\n", config);
+
+    if (enable) {
+        config |= bit;
+    }
+    else {
+        config &= ~bit;
+    }
+
+    DEBUG("[ds75lx] configuration register value: 0x%02X\n", config);
+
+    if (i2c_write_reg(DEV_I2C, DEV_ADDR, DS75LX_REG_CONFIGURATION, config, 0) < 0) {
+        DEBUG("[ds75lx] error writing configuration register\n");
+        /* Release I2C device */
+        i2c_release(DEV_I2C);
+
+        return -DS75LX_ERR_I2C;
+    }
+
+    /* Release I2C device */
+    i2c_release(DEV_I2C);
+
+    return DS75LX_OK;
+}
+
+int ds75lx_init(ds75lx_t *dev, const ds75lx_params_t *params)
+{
+    dev->params = *params;
+
+    /* Acquire exclusive access */
+    i2c_acquire(DEV_I2C);
+
+    uint8_t config = (dev->params.resolution << DS75LX_CONF_R0_POS);
+    /* force shutdown of the sensor */
+    config |= (1 << DS75LX_CONF_SD_POS);
+
+    DEBUG("[ds75lx] configuration register value: 0x%02X\n", config);
+
+    if (i2c_write_reg(DEV_I2C, DEV_ADDR, DS75LX_REG_CONFIGURATION, config, 0) < 0) {
```

Comment:
can't you use your `_set_configuration_bit` helper here, too?

Reply:
That might be possible, I'll give it a try.

Reply:
I pushed an update for this that should work (not tested yet). It saves 20B of ROM compared to the initial version.